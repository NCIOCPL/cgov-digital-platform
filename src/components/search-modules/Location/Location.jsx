import React, { useState } from 'react';
import { Fieldset, TextInput, Radio, Toggle, Dropdown } from '../../atomic';
import './Location.scss';

const Location = () => {
  const [activeRadio, setActiveRadio] = useState('search-location-all');
  const [limitToVA, setLimitToVA] = useState(false);

  const handleToggleChange = e => {
    setLimitToVA(e.target.checked);
  };

  const handleRadioChange = e => {
    setActiveRadio(e.target.value);
  };

  return (
    <Fieldset
      id="location"
      legend="Location"
      helpUrl="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#location"
      classes="search-location"
    >
      <p>
        Search for trials near a specific zip code; or in a country, state and
        city; or at a particular institution. The default selection will search
        for trials in all available locations.
      </p>
      <div className="data-toggle-block">
        <Toggle
          id="search-location-toggle"
          label="Limit results to Veterans Affairs facilities"
          onChange={handleToggleChange}
        />
        Limit results to Veterans Affairs facilities
      </div>
      <div className="group-locations">
        <Radio
          onChange={handleRadioChange}
          id="search-location-all"
          label="Search All Locations"
        />
        <Radio
          onChange={handleRadioChange}
          id="search-location-zip"
          label="ZIP Code"
        />
        {activeRadio === 'search-location-zip' && (
          <div className="search-location__block search-location__zip">
            <div className="--two-col">
              <TextInput
                id="search-location-zip-input"
                classes="zip"
                label="U.S. ZIP Code"
                maxLength={5}
              />
              <Dropdown
                id="search-location-radius"
                classes="radius"
                label="Radius"
                value={100}
              >
                {[20, 50, 100, 200, 500].map(dist => {
                  return (
                    <option key={dist} value={dist}>{`${dist} miles`}</option>
                  );
                })}
              </Dropdown>
            </div>
          </div>
        )}
        <Radio
          onChange={handleRadioChange}
          id="search-location-country"
          label="Country, State, City"
        />
        {activeRadio === 'search-location-country' && (
          <div className="search-location__block search-location__country">
            <Dropdown classes="country" label="Country">
              {[
                'United States',
                'United Kingdom',
                'Zambia',
                'Japan',
                'Uruguay',
              ].map(city => {
                return <option key={city} value={city}>{`${city}`}</option>;
              })}
            </Dropdown>
            <div className="search-location__country --two-col">
              <TextInput
                id="search-location-state"
                classes="state"
                label="State"
              />
              <TextInput
                id="search-location-city"
                classes="city"
                label="City"
              />
            </div>
          </div>
        )}
        {!limitToVA && (
          <>
            <Radio
              onChange={handleRadioChange}
              id="search-location-hospital"
              label="Hospitals/Institutions"
            />
            {activeRadio === 'search-location-hospital' && (
              <div className="search-location__block">
                <TextInput
                  id="hos"
                  label="Hospitals/Institutions"
                  labelHidden
                  placeHolder="Please enter 3 or more characters"
                />
              </div>
            )}
            <Radio
              onChange={handleRadioChange}
              id="search-location-nih"
              label="At NIH (only show trials at the NIH clinical center in Bethesda, MD)"
            />
          </>
        )}
      </div>
    </Fieldset>
  );
};

export default Location;
