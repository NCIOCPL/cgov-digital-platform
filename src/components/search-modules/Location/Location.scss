.search-location {
  .group-locations {
    padding-left: 1em;
  }
  &__block {
    margin: 0.5rem 0 1.5rem 2rem;

    .two-col {
      @media screen and (min-width: 641px) {
        display: flex;
        flex-direction: row;
        flex-wrap: nowrap;
        justify-content: space-between;
        > div {
          flex: 0 1 48.5%;
        }
      }
    }

    .zip,
    .country,
    .state {
      margin-bottom: 1rem;
    }

    .state,
    .zip {
      @media screen and (min-width: 641px) {
        margin-bottom: 0;
      }
    }

    .city,
    .state {
      width: auto;
      @media screen and (min-width: 641px) {
        width: 100%;
      }
    }
  }

  .--radius {
    margin-top: 1em;
    @media screen and (min-width: 641px) {
      margin-top: 0;
    }
  }
}
