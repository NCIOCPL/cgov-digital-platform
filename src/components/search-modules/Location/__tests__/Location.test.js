import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import { render } from 'enzyme';
import Location from '../Location';

const mockStore = configureMockStore();

const defaultProps = {};

const initialState = {
  form: {
    z: '',
    zp: '',
    lcnty: '',
    lcty: '',
    lst: '',
    hos: '',
  },
  cache: {
    countries: [],
  },
};

const setup = (enzymeMethod = render, props = {}, initialState = {}) => {
  const store = mockStore(initialState);
  const component = enzymeMethod(
    <Provider store={store}>
      <Location {...props} />
    </Provider>
  );
  return component;
};

describe('Location', () => {
  describe('Render', () => {
    it('renders without error', () => {
      const component = setup(render, defaultProps, initialState);
      expect(component).toMatchSnapshot();
    });
  });
});
