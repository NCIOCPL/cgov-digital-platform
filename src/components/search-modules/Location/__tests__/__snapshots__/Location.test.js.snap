// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Render renders without error 1`] = `
<fieldset
  class="cts-fieldset search-location"
  id="fieldset--location"
>
  <legend
    class="cts-fieldset__legend"
  >
    <span>
      Location
    </span>
    <a
      aria-label="Help"
      class="text-icon-help"
      href="https://www.cancer.gov/about-cancer/treatment/clinical-trials/search/help#location"
      rel="noopener noreferrer"
      target="_blank"
    >
      ?
    </a>
  </legend>
  <div
    class="cts-fieldset__body"
  >
    <p>
      Search for trials near a specific zip code; or in a country, state and city; or at a particular institution. The default selection will search for trials in all available locations.
    </p>
    <div
      class="data-toggle-block"
    >
      <div
        class="cts-toggle "
      >
        <input
          class="cts-toggle__input"
          id="search-location-toggle"
          name="search-location-toggle"
          type="checkbox"
        />
        <label
          aria-label="Limit results to Veterans Affairs facilities"
          class="cts-toggle__label"
          for="search-location-toggle"
        >
          <span
            aria-hidden="true"
            class="neg"
          >
            No
          </span>
          <span
            aria-hidden="true"
            class="pos"
          >
            Yes
          </span>
        </label>
      </div>
      Limit results to Veterans Affairs facilities
    </div>
    <div
      class="group-locations"
    >
      <div
        class="cts-radio "
      >
        <input
          aria-disabled="false"
          checked=""
          class="cts-radio__input"
          id="search-location-all"
          name="radios"
          type="radio"
          value="search-location-all"
        />
        <label
          class="cts-radio__label"
          for="search-location-all"
        >
          Search All Locations
        </label>
      </div>
      <div
        class="cts-radio "
      >
        <input
          aria-disabled="false"
          class="cts-radio__input"
          id="search-location-zip"
          name="radios"
          type="radio"
          value="search-location-zip"
        />
        <label
          class="cts-radio__label"
          for="search-location-zip"
        >
          ZIP Code
        </label>
      </div>
      <div
        class="cts-radio "
      >
        <input
          aria-disabled="false"
          class="cts-radio__input"
          id="search-location-country"
          name="radios"
          type="radio"
          value="search-location-country"
        />
        <label
          class="cts-radio__label"
          for="search-location-country"
        >
          Country, State, City
        </label>
      </div>
      <div
        class="cts-radio "
      >
        <input
          aria-disabled="false"
          class="cts-radio__input"
          id="search-location-hospital"
          name="radios"
          type="radio"
          value="search-location-hospital"
        />
        <label
          class="cts-radio__label"
          for="search-location-hospital"
        >
          Hospitals/Institutions
        </label>
      </div>
      <div
        class="cts-radio "
      >
        <input
          aria-disabled="false"
          class="cts-radio__input"
          id="search-location-nih"
          name="radios"
          type="radio"
          value="search-location-nih"
        />
        <label
          class="cts-radio__label"
          for="search-location-nih"
        >
          At NIH (only show trials at the NIH clinical center in Bethesda, MD)
        </label>
      </div>
    </div>
  </div>
</fieldset>
`;
